lib/services/elasticsearch.js

```diff
@@ -153,519 +166,479 @@ class ElasticSearch extends Service {
 
   /**
    * Get the document with given ID
-   * @param {Request} request
-   * @returns {Promise} resolve the document
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {string} id - Document ID
+   *
+   * @returns {Promise} { _id, _version, _source }
    */
-  get(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.id = request.input.resource._id;
+  get (index, collection, id) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      id
+    };
 
     // Just in case the user make a GET on url /mainindex/test/_search
     // Without this test we return something weird: a result.hits.hits with all document without filter because the body is empty in HTTP by default
     if (esRequest.id === '_search') {
-      return Bluebird.reject(this.getError('wrong_get_action'));
+      return this.reject('wrong_get_action');
     }
 
     return this.client.get(esRequest)
-      .then(result => {
-        if (
-          result._source
-          && result._source._kuzzle_info
-          && !result._source._kuzzle_info.active
-          && !request.input.args.includeTrash
-        ) {
-          return this.reject('document_not_found', esRequest.id);
-        }
+      .then(({ body }) => {
+        const apiResponse = {
+          _id: body._id,
+          _version: body._version,
+          _source: body._source
+        };
 
-        return prepareMetadata(result);
+        return apiResponse;
       })
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Return the list of documents matching the ids given in the body param
    * NB: Due to internal Kuzzle mechanism, can only be called on a single index/collection,
    * using the body { ids: [.. } syntax.
-   * @param {Request} request
-   * @returns {Promise}
+   *
+   * @param {string} index - Index name
+   * @param {string} collection - Collection name
+   * @param {Array<string>} ids - Document IDs
+   *
+   * @returns {Promise} { hits, total }
    */
-  mget(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.body = request.input.body;
-    delete esRequest.body.includeTrash;
+  mget (index, collection, ids) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: ids
+    };
 
     return this.client.mget(esRequest)
-      .then(result => {
-        // harmonize response format based upon the search one
-        if (result.docs) {
-          result.hits = request.input.args.includeTrash
-            ? result.docs
-            : result.docs.filter(
-              doc => doc._kuzzle_info === undefined ||
-              doc._kuzzle_info.active
-            );
-          delete result.docs;
-
-          result.hits = result.hits.map(obj => prepareMetadata(obj));
-        }
+      .then(({ body }) => {
+        const apiResponse = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "apiResponse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)